src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -1520,7 +1520,7 @@ bool Pet::removeSpell(uint32 spell_id, bool learn_prev, bool clear_ab)
     }
 
     // if remove last rank or non-ranked then update action bar at server and client if need
-    if (m_charmInfo->RemoveSpellFromActionBar(spell_id) && !learn_prev && clear_ab)
```

Comment:
> order matters in short-circuit boolean evaluation

That's the point. You're changing the order of the `RemoveSpellFromActionBar` invocation w.r.t. two different bools (`clear_ab` and `learn_prev`) - which of the two is the intended change (or is it both).